File: Avaliacoes/Java/Modulo1/src/main/java/controller/EventoController.java
package controller;

import model.SaldoCentral;
import thread.CaixaThread;
import view.CaixaView;

import java.util.ArrayList;
import java.util.List;

/**
 * Responsável por criar as 5 threads de caixa, iniciá-las, aguardar o término de todas e exibir o
 * resultado final através da View.
 *
 * @author Andrisa Santos
 */
public class EventoController {

    private SaldoCentral saldoCentral;
    private CaixaView view;
    private List<CaixaThread> caixas;

    public EventoController(SaldoCentral saldoCentral, CaixaView view) {
        this.saldoCentral = saldoCentral;
        this.view = view;
        this.caixas = new ArrayList<>();
    }

    public SaldoCentral getSaldoCentral() {
        return saldoCentral;
    }

    public void setSaldoCentral(SaldoCentral saldoCentral) {
        this.saldoCentral = saldoCentral;
    }

    public CaixaView getView() {
        return view;
    }

    public void setView(CaixaView view) {
        this.view = view;
    }

    /**
     *
     * @param numeroCaixas     quantidade de caixas/threads (5, pelo enunciado)
     * @param fichasPorCaixa   quantidade de fichas que cada caixa vende (1000)
     * @param precoFicha       preço de cada ficha (R$ 10,00)
     * @throws InterruptedException se a thread principal for interrompida
     */
    public void iniciarEvento(int numeroCaixas, int fichasPorCaixa, double precoFicha) throws InterruptedException {
        view.exibirCabecalho(numeroCaixas, fichasPorCaixa, precoFicha);

        for (int i = 1; i <= numeroCaixas; i++) {
            caixas.add(new CaixaThread("Caixa-" + i, saldoCentral, fichasPorCaixa, precoFicha));
        }

        for (CaixaThread c : caixas) c.start();      // fork
        for (CaixaThread c : caixas) {
            c.join();                                 // join
            view.exibirCaixaFinalizado(c.getName());
        }

        double esperado = numeroCaixas * fichasPorCaixa * precoFicha;
        view.exibirResultado(saldoCentral, esperado);
    }
}

File: Avaliacoes/CSharp/Modulo1/controller/EventoController.cs
using System.Collections.Generic;

/// <summary>
/// Cria as 5 threads de caixa,
/// inicia todas, aguarda o término de todas e exibe o resultado final.
/// </summary>
public class EventoController
{
    private SaldoCentral _saldoCentral;
    private CaixaView _view;
    private readonly List<CaixaThread> _caixas = new List<CaixaThread>();

    public EventoController(SaldoCentral saldoCentral, CaixaView view)
    {
        _saldoCentral = saldoCentral;
        _view = view;
    }

    public SaldoCentral SaldoCentral
    {
        get => _saldoCentral;
        set => _saldoCentral = value;
    }

    public CaixaView View
    {
        get => _view;
        set => _view = value;
    }

    /// <summary>
    /// Fluxo completo da simulação: cria as threads (fork), inicia todas,
    /// aguarda o término de todas (join) e exibe o resultado final.
    /// Note que não há "throws InterruptedException": C# não tem exceções
    /// checadas, então o método nem precisa declarar isso.
    /// </summary>
    public void IniciarEvento(int numeroCaixas, int fichasPorCaixa, double precoFicha)
    {
        _view.ExibirCabecalho(numeroCaixas, fichasPorCaixa, precoFicha);

        for (int i = 1; i <= numeroCaixas; i++)
        {
            _caixas.Add(new CaixaThread($"Caixa-{i}", _saldoCentral, fichasPorCaixa, precoFicha));
        }

        foreach (var c in _caixas)
        {
            c.Start(); // fork
        }

        foreach (var c in _caixas)
        {
            c.Join(); // join
            _view.ExibirCaixaFinalizado(c.Nome);
        }

        double esperado = numeroCaixas * fichasPorCaixa * precoFicha;
        _view.ExibirResultado(_saldoCentral, esperado);
    }
}
